src/config/mod.rs

```diff
@@ -324,12 +394,13 @@ impl Config {
                 err.push_str(msg)
             }
         }
-        match parsed.try_into() {
+
+        match parsed.try_into::<PartialConfig>() {
             Ok(parsed_config) => {
                 if !err.is_empty() {
                     eprint!("{err}");
                 }
-                Ok(Config::default().fill_from_parsed_config(parsed_config, dir))
+                Ok(parsed_config.to_parsed_config(style_edition, edition, version, dir))
```

Comment:
I also didn't love the `parsed_config.to_parsed_config` piece here, as the idents make for a confusing combination. I think this could be avoided with a simple rename but as always naming things is hard :shrug:

Reply:
Yeah, I saw that and thought the working was a bit confusing, but nothing too bad to stop us from moving forward with these changes.